### PR TITLE
feat(components): added aria-level attributes for dialog and drawer headers; Fixed input-number's aria-valuenow being null

### PR DESCRIPTION
**Pull request by Karolis-Stoncius**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### <samp>🤖 Generated by Copilot at 41ee964</samp>

This pull request adds a new prop `headerAriaLevel` to the `dialog` and `drawer` components, which allows customizing the heading level of the title in the `dialog-content` component. This improves the accessibility and semantics of the dialog and drawer components.

## Related Issue

Fixes #\_\_\_.

### <samp>🤖 Generated by Copilot at 41ee964</samp>

*  Add `headerAriaLevel` prop to `dialog` and `drawer` components to customize the heading level of the title ([link](https://github.com/element-plus/element-plus/pull/13822/files?diff=unified&w=0#diff-8e4884ffc45253766cb46796a84d2aeb1fb07f95966b3a4e2440f264948807b6R98-R104), [link](https://github.com/element-plus/element-plus/pull/13822/files?diff=unified&w=0#diff-e7f9a1cad1ef483c59fff16ce2adeb8cc10ab6e94338e776c75bf5d82a3d450eR24-R27))
*  Pass `headerAriaLevel` prop from `dialog` and `drawer` components to `dialog-content` component as `ariaLevel` prop ([link](https://github.com/element-plus/element-plus/pull/13822/files?diff=unified&w=0#diff-e9f41878502088bc1f6f1f49029526fcd3fa71df716e6b4dd199947c046b469dR49), [link](https://github.com/element-plus/element-plus/pull/13822/files?diff=unified&w=0#diff-130720e0fc0bcf5bbf918178a0f61269bd0a59071713fba0ff9b205d74726cfeR50))
*  Add `ariaLevel` prop to `dialog-content` component to set the header's aria-level attribute ([link](https://github.com/element-plus/element-plus/pull/13822/files?diff=unified&w=0#diff-b1abdb1afe933b28864280eb5eab6d5d9edc5174b3a04d819515ee9f015b3befR47-R53), [link](https://github.com/element-plus/element-plus/pull/13822/files?diff=unified&w=0#diff-bb6d76682592cb7b9dce997ae9cce0aaad06c8c422fda86008ab31c3d34f0756L5-R5))
*  Modify `aria-valuenow` attribute of the inner input element in `input-number` component to reflect the actual value of the input ([link](https://github.com/element-plus/element-plus/pull/13822/files?diff=unified&w=0#diff-0ae90cff73c6316f25336e807f41c1ecaca3dfbe4210f80273a4310aa89ac5a8L308-R311), [link](https://github.com/element-plus/element-plus/pull/13822/files?diff=unified&w=0#diff-0ae90cff73c6316f25336e807f41c1ecaca3dfbe4210f80273a4310aa89ac5a8L320-R323))

### CODE REVIEW

**Comment by Karolis-Stoncius** [comment]
Can anyone review this @btea @ryuhangyeong @chenxch ?

File: packages/components/input-number/src/input-number.vue [hunk 1]
```diff
@@ -317,7 +320,7 @@ onMounted(() => {
 })
 onUpdated(() => {
   const innerInput = input.value?.input
-  innerInput?.setAttribute('aria-valuenow', `${data.currentValue}`)
+  innerInput?.setAttribute('aria-valuenow', `${data.currentValue || ''}`)
```
Comment (btea) [inline packages/components/input-number/src/input-number.vue:323]:
```suggestion
  innerInput?.setAttribute('aria-valuenow', `${data.currentValue ?? ''}`)
```
The value of `data.currentValue` may be `0`.

**Comment by Karolis-Stoncius** [comment]
> @Karolis-Stoncius Thanks for the contribution! However, the team members are busy recently, so the review progress may be slower.
> 
> If you are in a hurry to use the functions you added, maybe you can modify and package them based on the current source code, and then use them in your project.
> 
> This PR seems to cause the test case to fail, please check it out.

Thank you for reviewing. I fixed your comment and updated the test

**Review by btea — APPROVED** [review]
LGTM

File: packages/components/dialog/src/dialog-content.ts [hunk 2]
```diff
@@ -44,6 +44,13 @@ export const dialogContentProps = buildProps({
     type: String,
     default: '',
   },
+  /**
+   * @description header's aria-level attribute
+   */
+  ariaLevel: {
+    type: String,
+    default: '1',
```
Comment (tolking) [inline packages/components/dialog/src/dialog-content.ts:52]: The default value is set to 2 or 3, which should be more reasonable. What do you think?

[with_heading_role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-level#with_heading_role)

File: packages/components/drawer/src/drawer.ts [hunk 3]
```diff
@@ -21,6 +21,10 @@ export const drawerProps = buildProps({
     type: Boolean,
     default: true,
   },
+  headerAriaLevel: {
+    type: String,
+    default: '1',
```
Comment (tolking) [inline packages/components/drawer/src/drawer.ts:26]: same

**Review by tolking — COMMENTED** [review]
It would be great if you could update the document at the same time

File: packages/components/dialog/src/dialog-content.ts (hunk 2 above)
Comment (Karolis-Stoncius) [reply packages/components/dialog/src/dialog-content.ts:52]: Good catch! [This article ](https://medium.com/@web-accessibility-education/which-heading-level-should-dialog-modals-have-7b3df89437f0)recommends using 2, so let's go with that If you're okay

Comment (tolking) [reply packages/components/dialog/src/dialog-content.ts:52]: I agree

**Comment by Karolis-Stoncius** [comment]
> It would be great if you could update the document at the same time

updated

File: docs/en-US/component/dialog.md [hunk 4]
```diff
@@ -139,6 +139,7 @@ When using `modal` = false, please make sure that `append-to-body` was set to **
 | destroy-on-close           | destroy elements in Dialog when closed                                                               | ^[boolean]                          | false   |
 | close-icon                 | custom close icon, default is Close                                                                  | ^[string] / ^[Component]            | —       |
 | z-index                    | same as z-index in native CSS, z-order of dialog                                                     | ^[number]                           | —       |
+| headerAriaLevel ^(a11y)    | header's `aria-level` attribute                                                                      | ^[number]                           | 2       |
```
Comment (tolking) [inline docs/en-US/component/dialog.md:142]:
```suggestion
| headerAriaLevel ^(a11y)    | header's `aria-level` attribute                                                                      | ^[string]                           | 2       |
```

Comment (Karolis-Stoncius) [reply docs/en-US/component/dialog.md]: fixed

File: docs/en-US/component/dialog.md [hunk 5]
```diff
@@ -139,6 +139,7 @@ When using `modal` = false, please make sure that `append-to-body` was set to **
 | destroy-on-close           | destroy elements in Dialog when closed                                                               | ^[boolean]                          | false   |
 | close-icon                 | custom close icon, default is Close                                                                  | ^[string] / ^[Component]            | —       |
 | z-index                    | same as z-index in native CSS, z-order of dialog                                                     | ^[number]                           | —       |
+| headerAriaLevel ^(a11y)    | header's `aria-level` attribute                                                                      | ^[string]                           | 2       |
```
Comment (tolking) [inline docs/en-US/component/dialog.md:142]:
```suggestion
| header-aria-level ^(a11y)    | header's `aria-level` attribute                                                                      | ^[string]                           | 2       |
```

Comment (Karolis-Stoncius) [reply docs/en-US/component/dialog.md]: Fixed, sorry I'm a bit distracted by my other projects 😅 Thanks for reviewing 🙇

**Review by tolking — APPROVED** [review]

**Comment by tolking** [comment]
Thank for your contribution

**Pull request merged**